javascripts/discourse/components/kanban/list.gjs

```diff
@@ -0,0 +1,348 @@
+      thisDefinition.params.assigned
+    ) {
+      let newUsername = thisDefinition.params.assigned;
+      if (newUsername === "*" || newUsername === "nobody") {
+        newUsername = undefined;
+      }
+      doUpdate = async () => {
+        const AssignModal =
+          require("discourse/plugins/discourse-assign/discourse/components/modal/assign-user").default;
+        this.modal.show(AssignModal, {
+          model: {
+            target: topic,
+            username: newUsername,
+            targetType: "Topic",
+            onSuccess: async () => {
+              this.refreshTopics();
+              oldRefresh();
+            },
+          },
+        });
+      };
+      requireConfirmation = false;
+    } else if (
+      thisDefinition.params.status === "closed" &&
+      oldDefinition.params.status === "open"
+    ) {
+      doUpdate = async () => {
+        await topic.saveStatus("closed", true);
+        this.refreshTopics();
+        oldRefresh();
+      };
+      confirmationMessage = I18n.t(themePrefix("confirm_close"), {
+        title: topic.title,
+      });
+    } else if (
+      thisDefinition.params.status === "open" &&
+      oldDefinition.params.status === "closed"
+    ) {
+      doUpdate = async () => {
+        await topic.saveStatus("closed", false);
+        this.refreshTopics();
+        oldRefresh();
+      };
+      confirmationMessage = I18n.t(themePrefix("confirm_open"), {
+        title: topic.title,
+      });
+    } else if (
+      thisDefinition.params.category &&
+      oldDefinition.params.category &&
+      thisDefinition.params.category !== oldDefinition.params.category
+    ) {
+      doUpdate = async () => {
+        try {
+          await Topic.update(topic, {
+            category_id: thisDefinition.params.category,
+            noBump: true,
+          });
+          this.refreshTopics();
+          oldRefresh();
+        } catch (error) {
+          popupAjaxError(error);
+        }
+      };
+      confirmationMessage = I18n.t(themePrefix("confirm_change_category"), {
+        title: topic.title,
+      });
+    }
+
+    if (requireConfirmation && confirmationMessage) {
+      this.dialog.yesNoConfirm({
+        message: confirmationMessage,
+        didConfirm: doUpdate,
+      });
+    } else {
+      await doUpdate();
+    }
+  }
+
+  @action
+  setDragData(data) {
+    data.oldDefinition = this.args.definition;
+    data.oldRefresh = () => this.refreshTopics();
+    this.args.setDragDataUpstream(data);
+  }
+
+  @action
+  calcHeight(element) {
+    const mainOutlet = element.closest("#main-outlet");
+    const mainOutletHeight = mainOutlet.getBoundingClientRect().height;
+    const mainOutletPadding = 40;
+    const listControlsHeight = mainOutlet
+      .querySelector(".list-controls")
+      .getBoundingClientRect().height;
+    const listTitleHeight = mainOutlet
+      .querySelector(".list-title")
+      .getBoundingClientRect().height;
+    const topicsHeight = document
```

Comment:
Unused variable? 

(I guess that means eslint isn't configured properly for gjs in this repo?)